guides/deploying-to-github-pages.md

```diff
@@ -0,0 +1,129 @@
+# Deploying to GitHub Pages
+
+GitHub Pages can host the static files that Tableau produces. This guide walks through a production-ready setup that you can adopt for your own site.
+
+## 1. Prepare your Tableau project
+
+1. Ensure your project exposes a `build` alias that calls `tableau.build`. The generator already includes this:
+
+   ```elixir
+   # mix.exs
+   defp aliases do
+     [
+       build: ["tableau.build"]
+     ]
+   end
+   ```
+
+2. Make sure `_site/` is gitignored. Tableau writes the built site there and your workflow will upload it as an artifact:
+
+   ```gitignore
+   /_site/
+   ```
+
+3. Configure your production URL in `config/prod.exs`. If you use a custom domain, set it to that value:
+
+   ```elixir
+   # config/prod.exs
+   config :tableau, :config,
+     url: "https://your-domain.tld"
+   ```
+
+   For GitHub Pages sites served from `<user>.github.io/<repo>`, use the full Pages URL and optionally set `base_path: "/<repo>"` so links render correctly locally.
+
+4. If you serve additional assets (Tailwind, esbuild, etc.), add the corresponding Mix tasks to the `build` alias so they also run in CI.
+
+## 2. Add the GitHub Pages workflow
+
+Create `.github/workflows/deploy.yml` with the following workflow. It installs Erlang/OTP + Elixir, builds the site in production mode, and publishes the `_site` directory to GitHub Pages:
+
+```yaml
+name: Deploy site to GitHub Pages
+
+on:
+  push:
+    branches: [main]
+
+permissions:
+  contents: read
+  pages: write
+  id-token: write
+
+concurrency:
+  group: pages
+  cancel-in-progress: true
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v5
+
+      - name: Set up Erlang/OTP and Elixir
+        uses: erlef/setup-beam@v1
+        with:
+          otp-version: "28"
+          elixir-version: "1.18"
+
+      - name: Cache Mix deps
+        uses: actions/cache@v4
+        with:
+          path: |
+            ~/.mix
```

Comment:
Is this necessary? I don't think I've ever seen this cached in GitHub actions before

Reply:
it makes the action run faster https://github.com/acrogenesis/acrogenesis.com/actions/runs/17779152948

Reply:
It makes it a bit faster it's where hex is stored and other archives.